app/uk/gov/hmrc/transitmovementsrouter/services/RoutingService.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsrouter.services
+
+import akka.NotUsed
+import akka.stream._
+import akka.stream.alpakka.xml.ParseEvent
+import akka.stream.alpakka.xml.scaladsl._
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.GraphDSL
+import akka.stream.scaladsl.Keep
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import com.google.inject._
+import play.api.Logging
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.transitmovementsrouter.connectors.MessageConnectorProvider
+import uk.gov.hmrc.transitmovementsrouter.models._
+import uk.gov.hmrc.transitmovementsrouter.services.error.RoutingError
+import cats.data.EitherT
+
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[RoutingServiceImpl])
+trait RoutingService {
+
+  def submitDeclaration(
+    movementType: MovementType,
+    movementId: MovementId,
+    messageId: MessageId,
+    payload: Source[ByteString, _]
+  )(implicit hc: HeaderCarrier): EitherT[Future, RoutingError, Unit]
+}
+
+class RoutingServiceImpl @Inject() (messageConnectorProvider: MessageConnectorProvider)(implicit materializer: Materializer)
+    extends RoutingService
+    with XmlParsingServiceHelpers
+    with Logging {
+
+  val office = Sink.head[ParseResult[OfficeOfDeparture]]
+
+  private val officeOfDepartureSink: Sink[ByteString, Future[ParseResult[OfficeOfDeparture]]] = Sink.fromGraph(
+    GraphDSL.createGraph(office) {
+      implicit builder => officeShape =>
+        import GraphDSL.Implicits._
+
+        val xmlParsing: FlowShape[ByteString, ParseEvent]                            = builder.add(XmlParsing.parser)
+        val officeOfDeparture: FlowShape[ParseEvent, ParseResult[OfficeOfDeparture]] = builder.add(XmlParser.officeOfDepartureExtractor)
+        xmlParsing ~> officeOfDeparture ~> officeShape
+
+        SinkShape(xmlParsing.in)
+    }
+  )
+
+  def buildMessage(messageId: MessageId): Flow[ByteString, ByteString, NotUsed] =
```

Comment:
Should be the `MessageSender`, not the `MessageId`.

Reply:
Agree.